c_binding/libsg.c

```diff
@@ -485,14 +486,14 @@ int _sg_io_vpd(char *err_msg, int fd, uint8_t page_code, uint8_t *data)
     int rc_vpd_00 = 0;
     char strerr_buff[_LSM_ERR_MSG_LEN];
     uint8_t sense_key = _T10_SPC_SENSE_KEY_NO_SENSE;
-    char sense_err_msg[_LSM_ERR_MSG_LEN];
+    char sense_err_msg[_LSM_ERR_MSG_LEN / 2];
     ssize_t data_len = 0;
 
     assert(err_msg != NULL);
     assert(fd >= 0);
     assert(data != NULL);
 
-    memset(sense_err_msg, 0, _LSM_ERR_MSG_LEN);
+    memset(sense_err_msg, 0, sizeof(sense_err_msg)/sizeof(char));
```

Comment:
`memset` clears a number of bytes, the statement `sizeof(sense_err_msg)/sizeof(char)` gives you the count or number of items stored in the memory which is not what what we want.  In this case we happen to be OK as `sizeof(char)` happens to be 1, thus N/1 == N.  Please simply change all the `memset` calls to use `sizeof(thing)`.

Reply:
I will fix it.